File: tests/csharp/csharp_oop/struct_basic.cs
// vybe-test: csharp/csharp_oop/struct_basic
// origin: languages/csharp/tests/csharp/test_csharp_oop.rs

using static __Harness;

var p = new Point(3, 4);
__P((p.Sum()).ToString());
__Check("7");

struct Point {
    public int X;
    public int Y;
    public Point(int x, int y) { X = x; Y = y; }
    public int Sum() { return X + Y; }
}

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/c/structs/struct_basic.c
// vybe-test: c/structs/struct_basic
// origin: languages/c/tests/c/test_structs.rs
#include <stdio.h>
#include <string.h>
#include <assert.h>

#include <stdio.h>
struct Point {
    int x;
    int y;
};
int main() {const char *__w[] = {"3\n", "4\n"};
int __n = 2, __i = 0;

    struct Point p;
    p.x = 3;
    p.y = 4;
    { char __t[512]; snprintf(__t, sizeof(__t), "%d\n", p.x);
  if (__i >= __n || strcmp(__t, __w[__i]) != 0) { printf("FAIL at line %d: got [%s]\n", __i, __t); assert(0); } __i++; }
    { char __t[512]; snprintf(__t, sizeof(__t), "%d\n", p.y);
  if (__i >= __n || strcmp(__t, __w[__i]) != 0) { printf("FAIL at line %d: got [%s]\n", __i, __t); assert(0); } __i++; }
    if (__i != __n) { printf("FAIL: %d line(s), wanted %d\n", __i, __n); assert(0); }
return 0;
}

